crates/ruff_python_formatter/src/statement/stmt_class_def.rs

```diff
@@ -23,40 +20,84 @@ impl FormatNodeRule<StmtClassDef> for FormatStmtClassDef {
             decorator_list,
         } = item;
 
-        f.join_with(hard_line_break())
-            .entries(decorator_list.iter().formatted())
-            .finish()?;
+        let comments = f.context().comments().clone();
+
+        let dangling_comments = comments.dangling_comments(item);
+        let trailing_definition_comments_start =
+            dangling_comments.partition_point(|comment| comment.line_position().is_own_line());
+
+        let (leading_definition_comments, trailing_definition_comments) =
+            dangling_comments.split_at(trailing_definition_comments_start);
+
+        if let Some(last_decorator) = decorator_list.last() {
+            f.join_with(hard_line_break())
+                .entries(decorator_list.iter().formatted())
+                .finish()?;
 
-        if !decorator_list.is_empty() {
-            hard_line_break().fmt(f)?;
+            if leading_definition_comments.is_empty() {
+                write!(f, [hard_line_break()])?;
+            } else {
+                // Write any leading definition comments (between last decorator and the header)
+                // while maintaining the right amount of empty lines between the comment
+                // and the last decorator.
+                let decorator_end =
+                    skip_trailing_trivia(last_decorator.end(), f.context().source());
```

Comment:
Nit: I find this name confusing. I would expect it to take an iterator as input (or something stateful that is after positioned after the trailing trivia). How about `find_first_non_trailing_trivia_position` or `first_non_trailing_trivia`

Related: There's a `token.kind().is_triva` method to avoid listing all trivia kinds.

Reply:
What was "Related 2" here?

Reply:
Whoops. That I was unsure if it should return an option

Reply:
:joy: